share/lib/python/neuron/expect_hocerr.py

```diff
@@ -20,19 +20,23 @@ def expect_hocerr(callable, args, sec=None):
   old_stderr = sys.stderr
   sys.stderr = my_stderr = StringIO()
   err = 0
+  pyerrmes = False
   try:
```

Comment:
Didn't think it out completely but it looks like this could be refactored into:

```python
try:
  # ...
except Exception as e:
  # Use `e` and `stderr` instead of `sys.exc_info`
else:
  # Do non-error stuff, if any
finally:
  # Restore stderr
```

Reply:
Thanks! That is very helpful.  I will modify according to your suggestion.This started out as a way to, without a lot of scaffolding, verify the hoc_execerror functionality and the possibility of recovery from such an error. This extends to the case where the error is actually raised by Python so I'm not sure that case really belongs here. Should there be a
```except_err(...)``` companion?  The other case that falls outside the domain of ```except_hocerr``` are non function call errors (i. e. statement errors) in internal getattr, setattr, etc. You can see in test_py2string.py the repetitious try
scaffolding for those cases. I have in the past used
```
cat e.py
import __main__
import sys
def e(stmt) :
  try:
    print(stmt)
    exec(stmt, __main__.__dict__)
  except:
    print((str(sys.exc_info()[0]) + ': ' + str(sys.exc_info()[1])))
```
which, as you remark, does not really need sys, but I couldn't generalize the context beyond ```__main__```.
It would be nice tor replace the try blocks with test like
```
expect_err('h.à = 1')
```
Let me know if you have a suggestion for that.